Dip/Dip/Dip.swift

```diff
@@ -340,14 +385,43 @@ extension DependencyContainer.Tag: StringLiteralConvertible {
   }
 }
 
+/*
+.String("a") == .Stirng("a")
+.Int(1) == .Int(1)
+.String("1") == .Int(1)
+*/
 public func ==(lhs: DependencyContainer.Tag, rhs: DependencyContainer.Tag) -> Bool {
   switch (lhs, rhs) {
   case let (.String(lhsString), .String(rhsString)):
     return lhsString == rhsString
   case let (.Int(lhsInt), .Int(rhsInt)):
     return lhsInt == rhsInt
-  default:
-    return false
+  case let (.String(lhsString), .Int(rhsInt)):
+    return lhsString == String(rhsInt)
+  case let (.Int(lhsInt), .String(rhsString)):
+    return String(lhsInt) == rhsString
+  }
+}
+
+/**
+ Tags are compared by comparing strings created from their assiciated values.
+ 
+ .String("a") < .String("b")
+ .Int(0) < .Int(1)
+ .String("0") < .Int(1)
+ .Int(0) < .String("1")
+ 
+ */
+public func <(lhs: DependencyContainer.Tag, rhs: DependencyContainer.Tag) -> Bool {
```

Comment:
This seems like a strange addition. I've not finished reviewing the PR yet so maybe I'm gonna see why this is needed later in the code, but I'm not sure why you need to compare tags and make them `Comparable` here. Checking for Equality makes sense, sorting tags, not so sure…?